File: Week2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count no of letter, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    // Print the grade level
    float l = (float) letters / words * 100;
    float s = (float) sentences / words * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);
    if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {

        printf("Grade %i\n", index);
    }
}
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }
    return letters;
}
int count_words(string text)
{
    int word = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            word++;
        }
    }
    return word;
}
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Week6/sentimental-readability/readability.py
from cs50 import get_string

# Get the text
text = get_string("Text: ")

# Count the letter
letter = 0
for char in text:
    if char.isalpha():
        letter += 1

# Count the word
word = 1
for char in text:
    if char == " ":
        word += 1

# Count the sentence
sentence = 0
for char in text:
    if char == "." or char == "!" or char == "?":
        sentence += 1

# Compute it
L = (letter / word) * 100
S = (sentence / word) * 100

aver = round(0.0588 * L - 0.296 * S - 15.8)

# Print the Grade
if aver < 1:
    print("Before Grade 1")
elif aver >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(aver))
